interfaces/matlab_experimental/Thermo/ThermoPhase.m

```diff
@@ -0,0 +1,1460 @@
+            end
+
+            if ~tp.isIdealGas
+                error('Gas object must represent an ideal gas mixture.');
+            end
+
+            if ~ischar(element)
+                error('Element name must be of format character.');
+            end
+
+            % Calculate the elemental mass fraction in a gas object using
+            % the following equation:
+            %
+            %    elMassFrac = sum of nAtoms(k, m)*Mel(m)*Y(k)/mw(k)
+            %
+            % where nAtoms(k, m) is the number of atoms of element 'm', in
+            % species 'k'; Mel(m) is the atomic weight of element 'm'; Y(k)
+            % is the mass fraction of species 'k'; and mw(k) is the
+            % molecular weight of species 'k'.
+
+            n = tp.nSpecies;
+            spec = tp.speciesNames;
+            eli = tp.elementIndex(element);
+            M = tp.atomicMasses;
+            Mel = M(eli);
+            MW = tp.molecularWeights;
+            % Initialize the element mass fraction as zero.
+            elMassFrac = 0.0;
+            % Perform summation of elemental mass fraction over all species.
+            for i = 1:n
+                natoms(i) = tp.nAtoms(spec{i}, element);
+                yy(i) = tp.massFraction(spec{i});
+                elMassFrac = elMassFrac + (natoms(i) * Mel * yy(i)) / MW(i);
+            end
+
+        end
+
+        function mmw = get.meanMolecularWeight(tp)
+            mmw = ctFunc('thermo_meanMolecularWeight', tp.tpID);
+        end
+
+        function density = get.molarDensity(tp)
+            density = ctFunc('thermo_molarDensity', tp.tpID);
+        end
+
+        function mw = get.molecularWeights(tp)
+            nsp = tp.nSpecies;
+            yy = zeros(1, nsp);
+            pt = libpointer('doublePtr', yy);
+            ctFunc('thermo_getMolecularWeights', tp.tpID, nsp, pt);
+            mw = pt.Value;
+        end
+
+        function n = nAtoms(tp, species, element)
+            % Number of atoms of an element in a species.
+            %
+            % n = tp.nAtoms(k,m)
+            %
+            % :param tp:
+            %     Instance of class :mat:class:`ThermoPhase` (or another
+            %     object that derives from ThermoPhase)
+            % :param k:
+            %     String species name or integer species number
+            % :param m:
+            %     String element name or integer element number
+            % :return:
+            %     Number of atoms of element ``m`` in species ``k``.
+
+            if nargin ~= 3
+                error('Two input arguments required.')
+            end
+
+            if ischar(species)
+                k = tp.speciesIndex(species);
+            else k = species;
+            end
+
+            if ischar(element)
+                m = tp.elementIndex(element);
+            else m = element;
+            end
+
+            if k < 0 | m < 0
+                n = -1;
```

Comment:
This is carried over from the legacy toolbox where error handling was done in Matlab. This shouldn't be necessary in the new toolbox since an invalid species or element name should return value `npos` which throws a Cantera error through `ctGetErr`. However, currently the error message is blank for `thermo_speciesIndex` and `thermo_elementIndex`.

Reply:
Should this be handled differently here then?
https://github.com/Cantera/cantera/blob/ba76a7be0a77629cbf97faff27b692229b134157/src/clib/ct.cpp#L147-L154